Content.Shared/GameObjects/Components/Interactable/CuffedComponent.cs

```diff
@@ -0,0 +1,173 @@
+using Serilog;
+using Timer = Robust.Shared.Timers.Timer;
+
+namespace Content.Shared.GameObjects.Components.Interactable
+{
+    [RegisterComponent]
+    public class CuffedComponent : Component, IActionBlocker
+    {
+#pragma warning disable 649
+        [Dependency] private readonly ISharedNotifyManager _notifyManager;
+#pragma warning restore 649
+
+        public override string Name => "Cuffed";
+
+        public string HandcuffId = "Handcuffs";
+
+        public float uncuffTime = 5.0f;
+        public float breakoutTime = 30.0f;
+        private CancellationTokenSource _cancellationTokenSource;
+
+        private float interactRange;
+
+        private GridCoordinates startPosition;
+
+        #region ActionBlockers
+        bool IActionBlocker.CanInteract() => false;
+
+        bool IActionBlocker.CanUse() => false;
+
+        bool IActionBlocker.CanPickup() => false;
+
+        bool IActionBlocker.CanDrop() => false;
+
+        bool IActionBlocker.CanAttack() => false;
+
+        bool IActionBlocker.CanEquip() => false;
+
+        bool IActionBlocker.CanUnequip() => false;
+
+        #endregion
+
+        public CuffedComponent()
+        {
+            interactRange = SharedInteractionSystem.InteractionRange / 2;
+            _cancellationTokenSource = new CancellationTokenSource();
+
+        }
+
+        public void TryUncuff(IEntity user, bool isOwner, bool force = false)
+        {
+            if (_cancellationTokenSource.Token != null)
+            {
+                _cancellationTokenSource.Cancel();
+            }
+            _cancellationTokenSource = new CancellationTokenSource();
+
+            if (!force)
+            {
+                if (!isOwner && !ActionBlockerSystem.CanInteract(user))
+                {
+                    return;
+                }
+                if (!isOwner && !EntitySystem.Get<SharedInteractionSystem>()
+                        .InRangeUnobstructed(user.Transform.MapPosition,
+                            Owner.Transform.MapPosition, interactRange, ignoredEnt: Owner))
+                {
+                    _notifyManager.PopupMessage(user, user,
+                        "You are too far away to remove the cuffs.");
+                    return;
+                }
+
+                startPosition = Owner.Transform.GridPosition;
+
+                _notifyManager.PopupMessage(user, user, "You start removing the cuffs.");
+                Logger.InfoS("fuck", "Here 1");
+                if (isOwner) // TODO: do_after() once it exists
+                {
+                    Logger.InfoS("fuck", "Here 2");
+                    Timer.Spawn(TimeSpan.FromSeconds(breakoutTime), () => Uncuff(user, true), _cancellationTokenSource.Token);
+                    Logger.InfoS("fuck", "Here 3");
+                }
+                else
+                {
+                    _notifyManager.PopupMessage(user, Owner, $"{user.Name} starts removing the cuffs.");
+                    Logger.InfoS("fuck", "Here 4");
```

Comment:
based

Reply:
If it's stupid and it works it's not stupid.

Reply:
exactly